fluffy/views.py

```diff
@@ -170,82 +171,141 @@ def upload():
 @app.route('/paste', methods={'POST'})
 def paste():
     """Paste and redirect."""
-    text = request.form['text']
-    # Browsers always send \r\n for the pasted text, which leads to bad
-    # newlines when curling the raw text (#28).
-    transformed_text = text.replace('\r\n', '\n')
+    lang = request.form['language']
+    objects = []
 
     with contextlib.ExitStack() as ctx:
-        objects = []
+        if lang == 'diff-between-two-texts':
+            text = request.form['diff']
+            text2 = request.form['diff2']
+            # Browsers always send \r\n for the pasted text, which leads to bad
+            # newlines when curling the raw text (#28).
+            transformed_text = text.replace('\r\n', '\n')
+            transformed_text2 = text2.replace('\r\n', '\n')
+            transformed_texts = [transformed_text, transformed_text2]
+            transformed_texts_len = [len(transformed_text.splitlines()), len(transformed_text2.splitlines())]
+
+            for txt in transformed_texts:
+                uf = _get_raw_text_object(txt, ctx)
+                objects.append(uf)
```

Comment:
This doesn't seem to be handling the possible error return value (the way it's written now, it can return a string error and HTTP status code, which are meant to be returned from the view to generate an HTTP 413 error):
https://github.com/chriskuehl/fluffy/blob/cbd1c44c9c4695c70eeed70ff4bdf947bd7ec8d9/fluffy/views.py#L269-L275